.vscode/tasks.json

```diff
@@ -114,13 +117,45 @@
       "problemMatcher": "$tsc"
     },
     {
-      "taskName": "Clean all generated files",
-      "isShellCommand": true,
+      "label": "Clean all generated files",
+      "group": "build",
+      "type": "shell",
       "command": "npm",
       "args": [
         "run",
         "clean"
       ]
+    },
+    {
+      "label": "Watch and compile source files",
+      "group": {
+        "kind": "build",
+        "isDefault": true
+      },
+      "command": "node",
+      "args": ["./bin/tsc-watch-monorepo"],
+      "isBackground": true,
+      "problemMatcher": "$tsc-watch",
+      "presentation": {
+        "reveal": "silent",
+        "focus": false,
+        "panel": "dedicated"
+      }
+    },
+    {
+      "label": "Run all tests",
+      "group": {
+        "kind": "test",
+        "isDefault": true
+      },
+      "type": "shell",
+      "command": "npm",
+      "args": ["run", "mocha", "-s"],
```

Comment:
Should linting be a part of this? Normally covered by `posttest` but we can't trigger `test` here since that builds and this is for watch right?

Reply:
Yes, this is for watch, thus we don't want to trigger `pretest` hook.

I am also intentionally not running any linter from this tasks, because typically linting is provided on the fly by Prettier/TSLint extensions installed in VS Code.

Sure, some linting problem won't be caught by those plugins (e.g. tslint rules requiring type info). To me, it's more important to have a fast red-green-refactor cycle that may miss some issues than to have a slow red-green-refactor cycle that always catches everything.

Here the workflow I am imagining:

 1. Open VSCode
 2. Start the watching build in background
 3. Write a failing tests, quickly run tests to see the new test fail (no linters)
 4. Make the new test pass, quickly run tests to verify
 5. Refactor, run tests to ensure nothing broke
 6. Rinse and repeat steps 2-5
 7. Run `npm test` from the command line
 8. Commit the changes

Note that our current watching build is not emitting `.d.ts` files, therefore it will not report missing imports needed by type definition files only (see e.g. https://github.com/Microsoft/TypeScript/issues/21673) - another reason why we should not rely on this "quick" mode and run `npm test` from time to time.